File: ruby/raindrops.rb
=begin
Write your code for the 'Raindrops' exercise in this file. Make the tests in
`raindrops_test.rb` pass.

To get started with TDD, see the `README.md` file in your
`ruby/raindrops` directory.
=end

class Raindrops
  def self.convert(number)
    if number % 3 === 0 && number % 5 === 0 && number % 7 === 0
      'PlingPlangPlong'
    elsif number % 3 === 0 && number % 5 === 0
      'PlingPlang'
    elsif number % 5 === 0 && number % 7 === 0
      'PlangPlong'
    elsif number % 7 === 0 && number % 3 === 0
      'PlingPlong'
    elsif number % 3 === 0
      'Pling'
    elsif number % 5 === 0
      'Plang'
    elsif number % 7 === 0
      'Plong'
    else
      "#{number}"
    end
  end
end

p Raindrops.convert(28)

File: python/raindrops.py
def convert(number):
  output = ''
  if number % 3 == 0 and number % 5 == 0 and number % 7 == 0:
    output = 'PlingPlangPlong'
  elif number % 3 == 0 and number % 5 == 0:
    output = 'PlingPlang'
  elif number % 5 == 0 and number % 7 == 0:
    output = 'PlangPlong'
  elif number % 7 == 0 and number % 3 == 0:
    output = 'PlingPlong'
  elif number % 3 == 0:
    output = 'Pling'
  elif number % 5 == 0:
    output = 'Plang'
  elif number % 7 == 0:
    output = 'Plong'
  else:
    output = "{}".format(number)

  return output


print(convert(15))